+bids/layout.m

```diff
@@ -456,66 +453,120 @@ function tolerant_message(use_schema, msg)
 
 function structure = manage_tsv(structure, pth, filename)
 
-  p = bids.internal.file_utils('FPList', pth,  ['^' strrep(filename, '.tsv', '\.tsv') '$']);
+  ext = bids.internal.file_utils(filename, 'ext');
+  tsv_file = bids.internal.file_utils('FPList', ...
+                                      pth,  ...
+                                      ['^' strrep(filename, ['.' ext], ['\.' ext]) '$']);
 
-  if isempty(p)
+  if isempty(tsv_file)
     warning('Missing: %s', fullfile(pth, filename));
 
   else
-    structure.content = bids.util.tsvread(p);
+    structure.content = bids.util.tsvread(tsv_file);
```

Comment:
Don't see much point to store content of tsv file in structure -- it can be large and their treatment is very much contextual.

Reply:
Yeah I was not super enthusiastic about this and it can potentially slow down the layout indexing if you have a large dataset.

Will remove it.

Reply:
OK I will start by removing the reading of the TSV content upon indexing for most modalities.

I am keeping some of it for `dwi` and `perf` **for now**. 

Most likely all tsv reading will be moved to the `query` side of things: will do that in several steps.

The good news is that tests run MUCH faster already.